src/docker/composeappengine.cc

```diff
@@ -168,7 +169,47 @@ bool ComposeAppEngine::isRunning(const App& app) const {
   return false;
 }
 
-std::string ComposeAppEngine::runningApps() const { return docker_client_->runningApps(); }
+Json::Value ComposeAppEngine::getRunningAppsInfo() const {
+  Json::Value apps;
+  try {
+    Json::Value containers;
+    docker_client_->getContainers(containers);
+
+    for (Json::ValueIterator ii = containers.begin(); ii != containers.end(); ++ii) {
+      Json::Value val = *ii;
+
+      std::string app_name = val["Labels"]["com.docker.compose.project"].asString();
+      if (app_name.empty()) {
+        continue;
+      }
+
+      App app{app_name, ""};
```

Comment:
seems like `App app` is only used for calling `state(app,..)`. Maybe just keep it scoped in that conditional block